### PR TITLE
HW-version1

### PR DESCRIPTION
**Pull request by stas83ggg**

### CODE REVIEW

**Review by okuzan — CHANGES_REQUESTED** [review]
Please make sure the build passes before requesting a review 🙏

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+                circle.setColor(colorSupplier.getRandomColor());
+                return circle;
+            case 1:
+                int at = random.nextInt(MAX_VALUE);
+                int bt = random.nextInt(MAX_VALUE);
+                int ht = random.nextInt(MAX_VALUE);
+                IsoscelesTrapezoid isoscelesTrapezoid = new IsoscelesTrapezoid(at, bt, ht);
+                isoscelesTrapezoid.setColor(colorSupplier.getRandomColor());
+                return isoscelesTrapezoid;
+            case 2:
```
Comment (aaananas) [inline src/main/java/core/basesyntax/FigureSupplier.java:26]:
```suggestion
                return getRandomCircle();
            case 1:
                return getRandomIsoscelesTrapezoid();
            case 2:
```
lets divide the logic into smaller private methods - it makes it easier to read your code

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                return square;
+        }
+    }
+
+    public Figure getDefaultFigure() {
+        int radius = 10;
+        Circle circle = new Circle(radius);
+        circle.setColor("WHITE");
+        return circle;
+    }
```
Comment (aaananas) [inline src/main/java/core/basesyntax/FigureSupplier.java:51]:
```suggestion
    public Figure getDefaultFigure() { 
        return new Circle(DEFAULT_RADIUS, DEFAULT_COLOR);
    }
```
avoid magic numbers ✨ 
use constants

File: src/main/java/core/basesyntax/Figure.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package core.basesyntax;
+
+public abstract class Figure implements Area, DrawArea {
+    private String color;
+
```
Comment (aaananas) [inline src/main/java/core/basesyntax/Figure.java:5]:
```suggestion
public abstract class Figure implements Area, DrawArea {
    private String color;

    public Figure(String color) {
        this.color = color;
    }
```

File: src/main/java/core/basesyntax/Circle.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
+package core.basesyntax;
+
+public class Circle extends Figure {
+    private final int radius;
+
+    public Circle(int radius) {
+        this.radius = radius;
+    }
```
Comment (aaananas) [inline src/main/java/core/basesyntax/Circle.java:8]:
```suggestion
    public Circle(String color, int radius) {
        super(color);
        this.radius = radius;
    }
```
you can call parent class constructor

**Review by aaananas — CHANGES_REQUESTED** [review]

**Comment by stas83ggg** [comment]
Will be new version

**Pull request closed without merging**